application_sandbox/overlapping_frames/main.cpp

```diff
@@ -187,515 +219,460 @@ vulkan::VulkanGraphicsPipeline buildTrianglePipeline(
+
+  pipeline.AddShader(VK_SHADER_STAGE_VERTEX_BIT, "main", postBuffer::vert);
+  pipeline.AddShader(VK_SHADER_STAGE_FRAGMENT_BIT, "main", postBuffer::frag);
+
+  pipeline.SetTopology(VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST);
+  pipeline.SetInputStreams(screen);
+  pipeline.SetScissor({
+      {0, 0},                                                // offset
+      {app->swapchain().width(), app->swapchain().height()}  // extent
+  });
+  pipeline.SetViewport({
+      0.0f,                                           // x
+      0.0f,                                           // y
+      static_cast<float>(app->swapchain().width()),   // width
+      static_cast<float>(app->swapchain().height()),  // height
+      0.0f,                                           // minDepth
+      1.0f                                            // maxDepth
+  });
+  pipeline.SetSamples(VK_SAMPLE_COUNT_1_BIT);
+  pipeline.AddAttachment();
+  pipeline.Commit();
+
+  return pipeline;
 }
 
 containers::vector<vulkan::ImagePointer> buildSamplerImages(
-    vulkan::VulkanApplication *app,
-    const entry::EntryData* data) {
-    containers::vector<vulkan::ImagePointer> images(data->allocator());
-    images.reserve(app->swapchain_images().size());
-
-    for(int index=0; index<app->swapchain_images().size(); index++) {
-        VkImageCreateInfo image_create_info = {
-            VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO,    // sType
-            nullptr,                                // pNext
-            0,                                      // flags
-            VK_IMAGE_TYPE_2D,                       // imageType
-            app->swapchain().format(),              // format
-            {
-                app->swapchain().width(),           // width
-                app->swapchain().height(),          // height
-                1                                   // depth
-            },                                      // extent
-            1,                                      // mipLevels
-            1,                                      // arrayLayers
-            VK_SAMPLE_COUNT_1_BIT,                  // samples
-            VK_IMAGE_TILING_OPTIMAL,                // tiling
-            VK_IMAGE_USAGE_SAMPLED_BIT |
-            VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT,    // usage
-            VK_SHARING_MODE_EXCLUSIVE,              // sharingMode
-            0,                                      // queueFamilyIndexCount
-            nullptr,                                // pQueueFamilyIndices
-            VK_IMAGE_LAYOUT_UNDEFINED               // initialLayout
-        };
-        images.push_back(app->CreateAndBindImage(&image_create_info));
-    }
-
-    return images;
+    vulkan::VulkanApplication* app, const entry::EntryData* data) {
+  containers::vector<vulkan::ImagePointer> images(data->allocator());
+  images.reserve(app->swapchain_images().size());
+
+  for (int index = 0; index < app->swapchain_images().size(); index++) {
```

Comment:
While we are making changes, can we switch this in 2 ways, 
1) size_t (since we are comparing with swapchain_images.size(), which is a size_t). Prevents a possible(unlikely) infinite loop
2) `size_t i = 0` to be more idiomatic C++

Reply:
done